### ISSUE TITLE
Failed to encrypt and send 1 OMEMO message(s) with: KEY_ERROR

### ISSUE THREAD

**Issue by o---** [UWPX/UWPX-Client#141]
## Bevor you create a new issue:

```
- [X] I searched for similar issues and did not find one
- [X] I'm using the latest version available in the [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
```

## I'm submitting a...:

- Bug report (I searched for similar issues and did not find one) 
 

## Current behavior:

First of all it took me a while to get OMEMO working at all. I was getting errors about OMEMO keys and removing and adding the account did not help. I had to completely uninstall the application. 

After that I managed to send and receive a couple of messages. Carbon copies did work too. Suddenly something happened. I am not sure if it was picture that I have received or what caused it, but all got messed up again. I dug out some messages. 

```
> [WARN][29.11.2020 12:52:21]: No need to navigate to page UWPX_UI.Pages.ChatPage - already on it. 

> [WARN][29.11.2020 12:56:41]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:178

> [ERROR][29.11.2020 12:57:14]: Failed to export logs: System.Exception: TEST EXEPTION
   at UWPX_UI.Pages.Settings.MiscSettingsPage.<ExportLogs_btn_Click>d__8.MoveNext() + 0x5a


> [WARN][29.11.2020 13:01:42]: Received not OMEMO encrypted message with a not matching public identity key from: contact1@domain.two:157 

> [ERROR][29.11.2020 14:19:19]: Failed to load OMEMO prekeys for: account1@domain.one

> [WARN][29.11.2020 14:19:19]: Only 0 found. Generating a new set. 
> [ERROR][30.11.2020 11:11:56]: OmemoStore: No such signed pre key: 5 
> [ERROR][30.11.2020 11:11:56]: Failed to load OMEMO signed prekey pair for: account1@domain.one
> [WARN][30.11.2020 11:12:10]: [XmppConnection]: Trying to connect but state is not DISCONNECTED or ERROR! State = CONNECTING 
> [ERROR][30.11.2020 11:12:21]: Failed to decrypt OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:12:48]: Failed to decrypt OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:12:55]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:13:16]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:13:28]: Failed to decrypt OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:14:49]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:15:23]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:15:41]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:16:52]: Failed to send OMEMO message - OmemoHelper is null 
> [ERROR][30.11.2020 11:17:01]: OmemoStore: No such signed pre key: 5 
> [ERROR][30.11.2020 11:17:01]: Failed to load OMEMO signed prekey pair for: account1@domain.one
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:178
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:178 
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:119 
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:119 
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:149 
> [WARN][30.11.2020 11:17:28]: Received not OMEMO encrypted message with a not matching public identity key from: account2@domain.two:149 
> [WARN][30.11.2020 11:17:29]: Received not OMEMO encrypted message with a not matching public identity key from: contact1@domain.two:157 
> [WARN][30.11.2020 11:18:33]: No need to navigate to page UWPX_UI.Pages.ChatPage - already on it. 
> [WARN][30.11.2020 11:25:46]: Received not OMEMO encrypted message with a not matching public identity key from: contact1@domain.two:157
> [WARN][30.11.2020 11:28:30]: Received not OMEMO encrypted message with a not matching public identity key from: contact1@domain.two:157 
```

## Expected behavior:

OMEMO is stable. 

## Minimal reproduction of the problem with instructions:
1. Activate OMEMO
2. Send a couple of messages and receive a ?picture? 

## Environment:

```
App Version(s): 
v.0.29.0.0 <!-- Which version of the App are you using (e.g. v.0.2.0) -->

Windows 10 Version Number: <!-- https://en.wikipedia.org/wiki/Windows_10_version_history -->
- [X] 1909
- [ ] 1809
- [ ] 1803
- [ ] 1709
- [ ] 1703
- [ ] 1607
- [ ] 1511
- [ ] 1507
- [ ] Insider Build (build number: )
- [ ] Misc:

Device form factor:
- [X] Desktop
- [ ] Mobile
- [ ] Xbox
- [ ] Surface Hub

Where did you got the APP from?
- [X] [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
- [ ] Self-build, using a provided release source
- [ ] Self-build, repo cloned at [dd.mm.yyy] <!-- When did you clone the repo! -->
- [ ] Misc, got it from... <!-- Please tell us your source! -->

```

**Comment by COM8** [comment]
This issue is related to loosing OMEMO Pre-keys from the DB.
My plan was to rewrite all the OMEMO-key-handling during my Christmas break.

Thanks for reporting this.

**Comment by rvsphinx** [comment]
i am having the same issue

![Screenshot 2021-02-04 150211](https://user-images.githubusercontent.com/1169695/106948710-0c59c900-66fa-11eb-8a55-0ecd4577f973.png)

**Comment by COM8** [comment]
#146 aims to fix all of those.

**Comment by COM8** [comment]
Fixed by #146.
